avm/res/app/job/main.bicep

```
@@ -51,11 +90,32 @@ param replicaRetryLimit int = 0
 param workloadProfileName string = 'Consumption'
 
 @description('Optional. The secrets of the Container App.')
-@secure()
-param secrets object?
+@metadata({
+  example: '''
+{
+  "name": "mysecret"
+  "identity": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myManagedIdentity",
+  "keyVaultUrl": "https://myvault${environment().suffixes.keyvaultDns}/secrets/mysecret",
+},
+{
+  "name": "mysecret"
+  "identity": "system",
+  "keyVaultUrl": "https://myvault${environment().suffixes.keyvaultDns}/secrets/mysecret",
+},
+{
+  "name": "mysecret",
+  "value": "mysecretvalue"
+}
+{
+  name: 'connection-string'
+  value: listKeys('/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/myResourceGroup/providers/Microsoft.Storage/storageAccounts/myStorageAccount', '2023-04-01').keys[0].value
+}
+'''
+})
+param secrets secretsType?
```

I see the most common practice is that of `<parameterNameInSingularForm>Type` e.g., for `volumeType`. Same for common extensions across the library, e.g., `diagnosticSettingType`, `privateEndpointType`. Hence here for consistency I'd suggest
```suggestion
param secrets secretType?
```